Diffuse/Shared/Diffuse.swift

```diff
@@ -65,4 +59,74 @@ public struct Diffuse<T> where T: Equatable {
         let allChanges = [insertedItems, removedItems, updatedItems, movedItems].flatMap { $0 }
         return CollectionChanges(allChanges: allChanges)
     }
+
+    public static func diff<T: Hashable>(old: [T], new: [T]) -> CollectionChanges {
+        // 1. Early return if either the new or the old array is empty.
+        if new.isEmpty {
+            let changes = old.enumerated().map({ offset, _ in return Change.remove(row: offset) })
+            return CollectionChanges(allChanges: changes)
+        } else if old.isEmpty {
+            let changes = new.enumerated().map({ offset, _ in return Change.insert(row: offset) })
+            return CollectionChanges(allChanges: changes)
+        }
+
+        // 2. Map both arrays to sets of `<Item<T>` type.
```

Comment:
```suggestion
        // 2. Map both arrays to sets of `Item<T>` type.
```
I guess this is a typo